### PR TITLE
Backports 0.15 pr10

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
~Each~Most of these backports was a clean cherry-pick, no conflicts were present. Not based on pr8/9

### CODE REVIEW

**Comment by UdjinM6** [comment]
`qt/sendcoinsdialog.cpp:207:13: error: 'class Ui::SendCoinsDialog' has no member named 'optInRBF'`

**Comment by PastaPastaPasta** [comment]
Removed one backport that was only relevant to RBF

File: test/functional/net.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply test/functional/net.py]: Let's just merge #2912 first if possible

File: src/Makefile.am [hunk 2]
```diff
@@ -638,6 +638,14 @@ config/stamp-h1: $(top_srcdir)/$(subdir)/config/dash-config.h.in $(top_builddir)
 $(top_srcdir)/$(subdir)/config/dash-config.h.in:  $(am__configure_deps)
 	$(AM_V_at)$(MAKE) -C $(top_srcdir) $(subdir)/config/dash-config.h.in
 
+
+config/bitcoin-config.h: config/stamp-h1
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/Makefile.am]: https://github.com/dashpay/dash/pull/2911/commits/111a573e15c4bae2d85675fc469b8e78ead7c86b

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -225,6 +225,37 @@ discussed extensively on the mailing list and IRC, be accompanied by a widely
 discussed BIP and have a generally widely perceived technical consensus of being
 a worthwhile change based on the judgement of the maintainers.
 
+### Finding Reviewers
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply CONTRIBUTING.md]: https://github.com/dashpay/dash/pull/2911/commits/5c199abd0207341ad1dd53fba23c0a29b3f8d461

**Comment by PastaPastaPasta** [comment]
Building locally

**Comment by PastaPastaPasta** [comment]
force pushed

**Comment by UdjinM6** [comment]
LGTM, let's wait for #2912

**Comment by PastaPastaPasta** [comment]
rebased

**Pull request merged**